File: framework/src/main/resources/templates/javascript/AppTest.js
var vertxTests = require('vertx_tests');
var vassert = require('vertx_assert');

var vertx = require('vertx');

var Yoke = require('yoke/Yoke');
var YokeTester = require('yoke/test/YokeTester');

// The test methods must begin with "test"

function testGetBase() {

    var yoke = new Yoke();
    // all resources are OK
    yoke.use(function (request) {
        request.response.end('OK');
    });

    new YokeTester(yoke).request('GET', '/', function(resp) {
        vassert.assertTrue(200 == resp.statusCode);
        vassert.testComplete();
    });
}

vertxTests.startTests(this);

File: framework/src/main/resources/templates/java/AppTest.java
import com.jetdrone.vertx.yoke.Yoke;
import com.jetdrone.vertx.yoke.Middleware;
import com.jetdrone.vertx.yoke.middleware.YokeRequest;
import com.jetdrone.vertx.yoke.test.Response;
import com.jetdrone.vertx.yoke.test.YokeTester;
import org.jetbrains.annotations.NotNull;
import org.junit.Test;
import org.vertx.java.core.Handler;
import org.vertx.testtools.TestVerticle;

import static org.vertx.testtools.VertxAssert.*;

public class AppTest extends TestVerticle {

    @Test
    public void testApp() {
        final Yoke yoke = new Yoke(this);
        yoke.use(new Middleware() {
            @Override
            public void handle(@NotNull YokeRequest request, @NotNull Handler<Object> next) {
                request.response().end("OK");
            }
        });

        final YokeTester yokeAssert = new YokeTester(yoke);

        yokeAssert.request("GET", "/", new Handler<Response>() {
            @Override
            public void handle(Response response) {
                assertEquals("OK", response.body.toString());
                testComplete();
            }
        });
    }
}
